### PR TITLE
Update the supported library set for Flutter for web

### LINKED ISSUE #34858: [Web] Dartev compile Error http module
hi im trying to run an existing flutter app (witch is ok for mobile and desktop) on web , (new way)
i have master version of flutter 

dartdev compile error http module , i tried with dart2js got exact same error
im a little new to this flutter thing , i appreciate a little light here 
flutter run -p web 
throws error
Screen Shot 2019-06-21 at 8 27 42 PM

pubspce : 

`name: myapp
version : 0.0.1
environment:
  sdk: ">=2.4.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter
  http: ^0.12.0+2
  equatable: ^0.2.4
  bloc: ^0.13.0
  flutter_bloc: ^0.13.0
  intl: ^0.15.8
  shared_preferences: 0.5.3+1
  build_runner: any
  build_web_compilers: ^2.1.1
  test: ^1.3.4
  cupertino_icons: ^0.1.2

dev_dependencies:
  flutter_test:
    sdk: flutter
flutter:
  uses-material-design: true

`

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

We're skipping build script checks for these libraries temporarily until we make a decision on dart:io support on the web. Reland of rolled back PR due to missing video_player.

EDIT:
In turns out I needed to land the refactor of goldens to not rely on platform specific code too ...

Includes some fixes from https://github.com/flutter/flutter/pull/38773

Fixes https://github.com/flutter/flutter/issues/34858

### CODE REVIEW

File: packages/flutter_tools/lib/src/build_runner/build_script.dart [hunk 1]
```diff
@@ -41,6 +41,19 @@ const String jsSourceMapExtension = '.ddc.js.map';
 const String kReleaseFlag = 'release';
 const String kProfileFlag = 'profile';
 
+// A minimum set of libraries to skip checks for to keep the examples compiling
+// until we make a decision on whether to support dart:io on the web.
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/build_runner/build_script.dart]: Is there a github issue for the dart:io decision?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_runner/build_script.dart]: Updated with the issue tracking the primary API decision for Platform

**Review by zanderso — APPROVED** [review]

File: packages/flutter_test/lib/src/_binding_io.dart [hunk 2]
```diff
@@ -0,0 +1,524 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:path/path.dart' as path;
```
Comment (zanderso) [inline packages/flutter_test/lib/src/_binding_io.dart]: nit: alphabetize imports

File: packages/flutter_test/lib/src/_goldens_io.dart [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+      + '_'
+      + failure
+      + '.png';
+    return File(_path.join('failures', testName));
+  }
+}
+
+/// Returns a [ComparisonResult] to describe the pixel differential of the
+/// [test] and [master] image bytes provided.
+ComparisonResult compareLists(List<int> test, List<int> master) {
```
Comment (zanderso) [inline packages/flutter_test/lib/src/_goldens_io.dart]: I haven't worked out all the details, but I suspect that with sufficient cleverness the analyzer and/or kernel compiler can be made to check that two different libraries implement the same signature by reducing it to the problem of checking that a class correctly implements an interface and then injecting the class into the right place, like we do in the tool. In a world where we want to do something like that, top-levels and statics would get in the way a bit.

Comment (jonahwilliams) [reply packages/flutter_test/lib/src/_goldens_io.dart]: I've implemented this check manually in the engine repo by walking both libraries are comparing types, its not great.

Currently we are running tests in both environments so we can catch type errors in CI.

Longer term I would like to not use config specific imports at all, but that is somewhat gated on what is decided on the SDK structure for flutter for web.

File: packages/flutter_test/lib/src/_binding_io.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_test/lib/src/_binding_io.dart]: Done

**Review by Piinks — APPROVED** [review]
LGTM!

**Pull request merged**